Sources/Imaging/Core/ImageReslice/index.js

```diff
@@ -32,68 +31,45 @@ function vtkImageReslice(publicAPI, model) {
   let optimizedTransform = null;
 
   function getImageResliceSlabTrap(tmpPtr, inComponents, sampleCount, f) {
-    let subTmpPtr = tmpPtr;
-    let m = inComponents;
     const n = sampleCount - 1;
-    do {
-      let result = subTmpPtr[0] * 0.5;
-      for (let k = n - 1; k !== 0; --k) {
-        subTmpPtr = tmpPtr.subarray(inComponents);
-        result += subTmpPtr[0];
+    for (let i = 0; i < inComponents; i += 1) {
+      let result = tmpPtr[i] * 0.5;
+      for (let j = 1; j < n; j += 1) {
+        result += tmpPtr[i + j];
```

Comment:
isn't it `tmpPtr[i + j*inComponents]` instead ?

Reply:
Yes. It must be modified.

Reply:
I modified it.